lib/Route.js

```diff
@@ -0,0 +1,272 @@
+
+    var routeDef = {
+        pattern: parsedPattern.pattern,
+        query: parsedPattern.query,
+        regex: parsedPattern.regex,
+        keys: parsedPattern.keys,
+        aliasResolver: aliasResolver
+    };
+
+    return new Route(routeDef, openApiDocs);
+};
+
+Route.prototype.getLegacyRouteObject = function () {
+    return this._routeDef;
+}
+
+Route.prototype.getSupportedProps = function (method) {
+    var handler = this.handlers[method] || [];
+    if (_.isFunction(handler)) {
+        return [];
+    }
+    return handler.reduce(function (acc, propHandler) {
+        acc.push.apply(acc, propHandler.props);
+        return acc;
+    }, []);
+};
+
+Route.prototype.isAlias = function () {
+    return !!this.aliasResolver;
+};
+
+Route.prototype.resolveAlias = function (request, connection) {
+    var target;
+    var resolver = this.aliasResolver;
+    if (typeof resolver === 'function') {
+        target = Promise.resolve(resolver(request, connection));
+    } else if (typeof resolver === 'string') {
+        request.setResourceId(resolver);
+        target = Promise.resolve(request);
+    }
+
+    return target;
+}
+
+Route.prototype.canHandleMethod = function (method) {
+    return this.handlers.hasOwnProperty(method)
+}
+
+Route.prototype.resolveHandlers = function (requestRouter, method) {
+    var handlers = [];
+    if (_.isFunction(this.handlers[method])) {
+        handlers.push(this.handlers[method]);
+    } else if (_.isArray(this.handlers[method])) {
+        var props = requestRouter.getProps();
+        var resource = requestRouter.getResource();
+
+        //For PUT OR PATCH map the resource passed in to the correct handler
+        if (typeof resource !== 'undefined' && [Request.METHOD_PATCH, Request.METHOD_PUT].indexOf(method) !== -1) {
+            var keys = Object.keys(resource);
+
+            //If resource passed in is an empty array, return an empty resource
+            if (!keys.length) {
+                return new Promise.resolve({});
+            }
+
+            handlers = this.getHandlers(requestRouter, keys);
+        } else if (props.length) {
+            handlers = this.getHandlers(requestRouter, props);
+        } else {
+            this.handlers[method].forEach(function (handler) {
+                handlers.push(handler.callback);
+            });
+        }
+    }
+
+    return handlers;
+}
+
+Route.prototype.getHandlers = function (request, input) {
+    var method = request.getMethod();
+    var handlers = [];
+
+    this.handlers[method].filter(function (handler) {
+        var topLevelInput = input.map(function (input) {
+            return input.split(/[\.@]/)[0];
+        });
+        return _.intersection(topLevelInput, handler.props).length;
+    }).forEach(function (handler) {
+        handlers.push(handler.callback);
+    });
+
+    return handlers;
+}
+
+Route.prototype.getOpenApiDocumentation = function () {
+    return new Promise(function (resolve, reject) {
```

Comment:
nit: nothing within this promise is async, so you could simplify this a bit by returning `Promise.resolve(value)`